cloudinary-core/src/main/java/com/cloudinary/Configuration.java

```diff
@@ -43,11 +46,32 @@
     public AuthToken authToken;
     public boolean forceVersion = true;
     public boolean longUrlSignature = DEFAULT_IS_LONG_SIGNATURE;
+    public Signer signatureAlgorithm = DEFAULT_SIGNATURE_ALGORITHM;
```

Comment:
This `Signer` class is a bit problematic - It's either a `signer` _or_ a `signing algorithm`. It can't be both.
We need a `SignatureAlgorithm` enum. The `Signer` class itself is a bit unnecessary. 

Whenever an algorithm is passed around or stored (e.g. in config) it needs to be the enum, not a signer class.